cmake/CompilerWarnings.cmake

```
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+function(set_default_compiler_warnings target_name)
+    option(WARNINGS_AS_ERRORS "Treat compiler warnings as errors" ON)
+
+    set(MSVC_WARNINGS
+        /W4 # Baseline reasonable warnings
+        /w14242 # 'identifier': conversion from 'type1' to 'type1', possible loss of data
```

[question] I don't quite get what's happening here. [This page](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4242?view=msvc-170) seems to suggest that `4242` is level 4. We've enabled level 4 above, but then this line is promoting this warning to Level 1 even thought it would be displayed anyway - by being Level 4?

Is `/W4` sufficient by itself, given the MS advice:

> We recommend that you use this option to provide lint-like warnings. For a new project, it may be best to use /W4 in all compilations. This option helps ensure the fewest possible hard-to-find code defects.

---

There is also a typo in the comment
```suggestion
        /w14242 # 'identifier': conversion from 'type1' to 'type2', possible loss of data
```

---

It's all a bit weird.
This may be a bit old, but it's what I always think of with warning level 4.
http://gina.xmmg.com/dpweb/docs/CompilingWarningLevel4.html

Which is why I advocate level 3.